### PR TITLE
HXS-2775: Impact Tracking

### PR DESCRIPTION
**Pull request by iamkarsoft**
Link to Ticket: [https://helixsleep.atlassian.net/browse/HXS-2775](https://helixsleep.atlassian.net/browse/HXS-2775)

This PR adds an input field for the admin to be able to enter Server side URL which will be injected into the code rather than hard coded.

### CODE REVIEW

**Review by BaffourAdu — APPROVED** [review]

**Review by garos4 — APPROVED** [review]

**Review by tonning — CHANGES_REQUESTED** [review]
I think instead of hardcoding the `https://` we should just let the user have to include that in the input for Server Side URL.

Also, let's add a helper text to the new input saying something along the lines of "(Optional) Uses GTM default URL if left blank."

**Review by OniiCoder — APPROVED** [review]

File: resources/views/body-script.blade.php [hunk 1]
```diff
@@ -1,6 +1,26 @@
...
+
+            if(! Str::startsWith($server_side_url,'https://')){
+                $server_side_url = Str::start($server_side_url,'https://');
+
+            }
+
+        }
+
+    @endphp
+
```
Comment (tonning) [inline resources/views/body-script.blade.php]: Now that you don't have to worry about have the URL start with `https://` you can just do an simple ternary.

```suggestion
    @php
        $serverSideUrl = blank(settings(Astrogoat\Gtm\Settings\GtmSettings::class, 'server_side_url'))
            ? 'https://www.googletagmanager.com'
            : settings(Astrogoat\Gtm\Settings\GtmSettings::class, 'server_side_url');
    @endphp
```

I re-formatted the variable to be camelCase for consistency.

**Review by tonning — CHANGES_REQUESTED** [review]

File: resources/views/body-script.blade.php (hunk 1 above)
Comment (iamkarsoft) [reply resources/views/body-script.blade.php]: maybe i was just trying to do too much! Thank you 😄

Comment (tonning) [reply resources/views/body-script.blade.php]: Yeah, sometimes less is more :) Don't forget to also update the other script file, then we should be good here 👍

**Review by tonning — APPROVED** [review]

**Pull request merged**